src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -977,25 +978,27 @@ private static void Parse(
 
                     numberOfRowsForValues++;
                     database.Append(tokenType, tokenStart, DbRow.UnknownSize);
-                    var row = new StackRow(numberOfRowsForMembers + 1);
+                    var row = new StackRow(numberOfProperties, numberOfRowsForMembers + 1);
                     stack.Push(row);
                     numberOfRowsForMembers = 0;
+                    numberOfProperties = 0;
                 }
                 else if (tokenType == JsonTokenType.EndObject)
                 {
                     int rowIndex = database.FindIndexOfFirstUnsetSizeOrLength(JsonTokenType.StartObject);
 
                     numberOfRowsForValues++;
                     numberOfRowsForMembers++;
-                    database.SetLength(rowIndex, numberOfRowsForMembers);
+                    database.SetLength(rowIndex, numberOfProperties);
 
                     int newRowIndex = database.Length;
                     database.Append(tokenType, tokenStart, reader.ValueSpan.Length);
                     database.SetNumberOfRows(rowIndex, numberOfRowsForMembers);
                     database.SetNumberOfRows(newRowIndex, numberOfRowsForMembers);
 
                     StackRow row = stack.Pop();
-                    numberOfRowsForMembers += row.SizeOrLength;
+                    numberOfRowsForMembers += row.NumberOfRows;
+                    numberOfProperties = row.SizeOrLength;
```

Comment:
Minor nit pick: assignments made in different order compared to the array counterpart.
```suggestion
                    numberOfProperties = row.SizeOrLength;
                    numberOfRowsForMembers += row.NumberOfRows;
```

Reply:
Sure! I also changed the order of another assignment - just to be consistent. See https://github.com/dotnet/runtime/pull/106503/commits/0c6fc09ba6639b351f7bba793c92add711f83227